Apromore-Custom-Plugins/Process-Discoverer-Portal-Plugin/src/main/java/org/apromore/plugin/portal/processdiscoverer/impl/apmlog/LogFilterControllerWithAPMLog.java

```diff
@@ -151,14 +152,21 @@ private Map<String, Object> getEditorOptionPayload(FilterType filterType,
         switch (filterType) {
             case CASE_SECTION_ATTRIBUTE_COMBINATION:
             case EVENT_ATTRIBUTE_DURATION:
+            case NODE_COST:
                 data = (String) param.get("data");
                 if (filterType == FilterType.EVENT_ATTRIBUTE_DURATION &&
                         !isValidAttributeDurationPayload(mainAttribute, data)) {
                     Messagebox.show(
                             parent.getLabel("failedFilterNodeDurationSingleValue_message"),
-                            parent.getLabel("common_error_text"), Messagebox.OK, Messagebox.ERROR);
+                            parent.getLabel(ERROR_TITLE_KEY), Messagebox.OK, Messagebox.ERROR);
                     return null;
                 }
+                if (filterType == FilterType.NODE_COST && parent.getProcessAnalyst().getRoleValues().isEmpty()) {
+                    Messagebox.show(
```

Comment:
Should check parent.getProcessAnalyst().getRoleValues().isEmpty() is enough, the filterType is being NODE_COST in this switch case.

Reply:
I see, this is for 3 switch cases, not one.

Reply:
This code is reached if filterType is either CASE_SECTION_ATTRIBUTE_COMBINATION,  EVENT_ATTRIBUTE_DURATION or NODE_COST so there is still a need to check for the filterType.

Reply:
The same comment, can we move the error message first and outside the switch case to make the switch...case block cleaner to read?